internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -177,25 +195,70 @@ func (r *Reconciler) reconcile(ctx context.Context, clusterClass *clusterv1.Clus
 		return ctrl.Result{}, kerrors.NewAggregate(errs)
 	}
 
-	// Ensure the variables are added to the ClusterClass status.
-	clusterClass.Status.Variables = []clusterv1.ClusterClassStatusVariable{}
-	for _, variable := range clusterClass.Spec.Variables {
-		clusterClass.Status.Variables = append(clusterClass.Status.Variables,
-			clusterv1.ClusterClassStatusVariable{
-				Name: variable.Name,
-				Definitions: []clusterv1.ClusterClassStatusVariableDefinition{
-					{
-						From:     clusterv1.VariableDefinitionFromInline,
-						Required: variable.Required,
-						Schema:   variable.Schema,
-					},
-				}})
+	// Reconcile variables
+	if errs = r.reconcileVariables(ctx, clusterClass); len(errs) > 0 {
+		return ctrl.Result{}, kerrors.NewAggregate(errs)
 	}
+
 	reconcileConditions(clusterClass, outdatedRefs)
 
 	return ctrl.Result{}, nil
 }
 
+func statusVariableFromClusterClassVariable(variable clusterv1.ClusterClassVariable, from string) clusterv1.ClusterClassStatusVariable {
+	return clusterv1.ClusterClassStatusVariable{
+		Name: variable.Name,
+		// TODO: In a future iteration this should be false where definitions are equal.
+		DefintionsConflict: true,
+		Definitions: []clusterv1.ClusterClassStatusVariableDefinition{
+			{
+				From:     from,
+				Required: variable.Required,
+				Schema:   variable.Schema,
+			},
+		}}
+}
+
+func (r *Reconciler) reconcileVariables(ctx context.Context, clusterClass *clusterv1.ClusterClass) []error {
+	errs := []error{}
+
+	// Add inline variable definitions to the ClusterClass status.
+	clusterClass.Status.Variables = []clusterv1.ClusterClassStatusVariable{}
+	for _, variable := range clusterClass.Spec.Variables {
+		clusterClass.Status.Variables = append(clusterClass.Status.Variables, statusVariableFromClusterClassVariable(variable, clusterv1.VariableDefinitionFromInline))
+	}
+
+	// If RuntimeSDK is enabled call the DiscoverVariables hook for all associated Runtime Extensions and add the variables
+	// to the ClusterClass status.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		for _, patch := range clusterClass.Spec.Patches {
+			if patch.External == nil || patch.External.DiscoverVariablesExtension == nil {
+				continue
+			}
+			req := &runtimehooksv1.DiscoverVariablesRequest{}
+			req.Settings = patch.External.Settings
+
+			resp := &runtimehooksv1.DiscoverVariablesResponse{}
+			err := r.RuntimeClient.CallExtension(ctx, runtimehooksv1.DiscoverVariables, clusterClass, *patch.External.DiscoverVariablesExtension, req, resp)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "failed to call DiscoverVariables for patch %s", patch.Name))
```

Comment:
```suggestion
				errs = append(errs, errors.Wrapf(err, "failed to discover variables for ClusterClass %s: failed to call DiscoverVariables for patch %s", clusterClass.Name, patch.Name))
```
nit: to end up with an error consistent to below

Reply:
Thinking about it some more. WDYT about adding the prefix (`failed to discover variables for ClusterClass %s`) only once and in l. 258? (and call aggregate already in l.258)